src/qibolab/transpilers/general_connectivity.py

```diff
@@ -0,0 +1,420 @@
+            final_mapping (dict): final qubit mapping.
+            init_mapping (dict): initial qubit mapping.
+            added_swaps (int): number of swap gates added.
+        """
+        self._circuit_position = 0
+        self._added_swaps = 0
+        self.translate_circuit(qibo_circuit)
+        keys = list(self._connectivity.nodes())
+        if self._init_method == "greedy":
+            self.greedy_init()
+        elif self._init_method == "subgraph":
+            self.subgraph_init()
+        elif self._init_method == "custom":
+            self._mapping = {keys[i]: self._mapping["q" + str(i)] for i in range(len(keys))}
+            self._graph = nx.relabel_nodes(self._connectivity, self._mapping)
+        else:
+            raise_error(ValueError, "Wrong qubit mapping initialization method (use 'greedy', 'subgraph' or 'custom')")
+        init_qubit_map = self.init_qubit_map()
+        init_mapping = {keys[i]: init_qubit_map[i] for i in range(len(keys))}
+        self._qubit_map = np.sort(init_qubit_map)
+        self.init_circuit(qibo_circuit)
+        len_2q_circuit = len(self._circuit_repr)
+        self._circuit_repr = self.reduce(self._graph)
+        matched_2q_gates = len_2q_circuit - len(self._circuit_repr)
+        self.add_gates(qibo_circuit, matched_2q_gates)
+        while len(self._circuit_repr) != 0:
+            len_2q_circuit = len(self._circuit_repr)
+            path, meeting_point = self.relocate()
+            matched_2q_gates = len_2q_circuit - len(self._circuit_repr)
+            self.add_swaps(path, meeting_point)
+            self.update_qubit_map()
+            self.add_gates(qibo_circuit, matched_2q_gates)
+        final_mapping = {keys[i]: init_qubit_map[self._qubit_map[i]] for i in range(len(keys))}
+        return (
+            self.init_mapping_circuit(self.transpiled_circuit, init_qubit_map),
+            final_mapping,
+            init_mapping,
+            self._added_swaps,
+        )
+
+    def custom_qubit_mapping(self, map):
+        """Define a custom initial qubit mapping.
+
+        Args:
+            map (dict): Dictionary of the type {q0: 1, q1: 2, q2: 0} reporting the circuit to chip qubit mapping.
+        """
+        self._init_method = "custom"
+        self._mapping = map
+
+    def connectivity(self, connectivity):
+        """Set the hardware chip connectivity.
+
+        Args:
+            connectivity (string or networkx graph): set a special TII chip connectivity (using a string) or define a
+            custom connectivity (using a networkx graph).
+        """
+        if isinstance(connectivity, str):
+            self._connectivity = self.special_connectivity(connectivity)
+        elif isinstance(connectivity, type(nx.Graph())):
+            self._connectivity = connectivity
+        else:
+            raise_error(TypeError, "Use string for special connectivity or networkx graph for custom connectivity")
```

Comment:
I think it may be useful to check if the connectivity is a networkx graph. Moreover in this way it is possible to perform tests with different connectivities without having to create a new transpiler. Anyway i have just changed this function into a setter method with Stavros